web/src/routes/package.js

```diff
@@ -112,8 +120,111 @@ package_router.post('/', async (req,res) => {
+                // check if a package with that URL already exists
+                else if(await PackageData.findOne({ URL: newPackageDataSchema.URL })) {
+                    res.status(409).json({ message: 'Package exists already.' })
+                }
+                else{
+                    // call the child process using the URL
+                    let rating_output = ""
+                    async function callRatingCLI()
+                    {
+                        const { stdout } = await execFile('./461_CLI/route_run', [newPackageDataSchema.URL]);
+                        rating_output = JSON.parse(stdout)
+                    }
+                    await callRatingCLI()
+                    
+                    // check if all ratings are above 0.5
+                    if(rating_output['BUS_FACTOR_SCORE'] < 0.5 || rating_output['CORRECTNESS_SCORE'] < 0.5 
+                        || rating_output['CORRECTNESS_SCORE'] < 0.5 || rating_output['RAMP_UP_SCORE'] < 0.5 
+                        || rating_output['RESPONSIVE_MAINTAINER_SCORE'] < 0.5 || rating_output['LICENSE_SCORE'] < 0.5
+                        || rating_output['VERSION_SCORE'] < 0.5 || rating_output['CODE_REVIEWED_PERCENTAGE'] < 0.5 )
+                    {
+                        res.status(424).json({ message: "Package not uploaded due to disqualified rating" })
+                    }
+                    else
+                    {
+                        // setting owner/repo args for calls to API
+                        let owner;
+                        let repo;
+                        if(url_elements.includes('github.com'))
+                        {
+                            owner = url_elements[3];
+                            repo = url_elements[4];
+                        }
+                        else // for npm URLs
+                        {
+                            owner = url_elements[4];
+                            const response = await axios.get(`https://registry.npmjs.org/${owner}`);
+                            url_out = response.data.repository['url'];
+                            url_parts = url_out.split("/");
+                            owner = url_parts[3];
+                            repo_withgit = url_parts[4];
+                            repo_parts = repo_withgit.split(".git");
+                            repo = repo_parts[0];
+                        }
+
+                        // Get the package.json using Github REST API and extract name and version from package.json
+                        const base64Encoded = await axios.get(`https://api.github.com/repos/${owner}/${repo}/contents/package.json`, {
+                            headers: {
+                                'Authorization': `Token ${environment.GITHUB_TOKEN}`
+                            }
+                        });
+                        let newName
+                        let newVersion
+                        let zipError = false
```

Comment:
Might be worth to change this variable name since it doesn't correlate to zip error anymore. PackageJSONError perhaps